services/base-noncaching.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const makeBadge = require('../gh-badges/lib/make-badge')
+const { makeSend } = require('../lib/result-sender')
+const BaseService = require('./base')
+const { setCacheHeaders } = require('./cache-headers')
+
+// All services may be cached downstream, as is controlled by the service,
+// the user's request, and the configured default cache length.
+//
+// The `handle()` function of most `BaseService` subclasses is wrapped in
+// _additional_  caching, which lives in memory on the servers. See
+// `lib/request-handler.js` and `BaseService.prototype.register()`.
+//
+// In contrast, services deriving from `BaseNoncachedService` are _not_ cached
+// on the server. This means that each request that hits the server triggers
+// another call to the handler. This makes them more useful for server
+// diagnostics.
+module.exports = class NoncachingBaseService extends BaseService {
+  static register({ camp }, serviceConfig) {
+    const { cacheHeaders: cacheHeaderConfig } = serviceConfig
+    const { _cacheLength: serviceCacheLengthSeconds } = this
```

Comment:
This is effectively generalizing the logic that came from the bit flip handler in `server.js`. I set `debug` services default `_cacheLength` to 60 seconds.